homeassistant/components/zwave_js/sensor.py

```diff
@@ -119,62 +252,16 @@ def __init__(
         self,
         config_entry: ConfigEntry,
         client: ZwaveClient,
-        info: ZwaveDiscoveryInfo,
+        entity_description: ZwaveSensorEntityDescription,
     ) -> None:
         """Initialize a ZWaveSensorBase entity."""
-        super().__init__(config_entry, client, info)
+        assert entity_description.info
+        super().__init__(config_entry, client, entity_description.info)
+        self.entity_description = entity_description
 
         # Entity class attributes
+        self._attr_force_update = True
         self._attr_name = self.generate_name(include_value_name=True)
-        self._attr_device_class = self._get_device_class()
-        self._attr_state_class = self._get_state_class()
-
-    def _get_device_class(self) -> str | None:
-        """
-        Get the device class of the sensor.
-
-        This should be run once during initialization so we don't have to calculate
-        this value on every state update.
-        """
-        if self.info.primary_value.command_class == CommandClass.BATTERY:
-            return DEVICE_CLASS_BATTERY
-        if isinstance(self.info.primary_value.property_, str):
-            property_lower = self.info.primary_value.property_.lower()
-            if "humidity" in property_lower:
-                return DEVICE_CLASS_HUMIDITY
-            if "temperature" in property_lower:
-                return DEVICE_CLASS_TEMPERATURE
-        if self.info.primary_value.metadata.unit == "A":
-            return DEVICE_CLASS_CURRENT
-        if self.info.primary_value.metadata.unit == "W":
-            return DEVICE_CLASS_POWER
-        if self.info.primary_value.metadata.unit == "kWh":
-            return DEVICE_CLASS_ENERGY
-        if self.info.primary_value.metadata.unit == "V":
-            return DEVICE_CLASS_VOLTAGE
-        if self.info.primary_value.metadata.unit == "Lux":
```

Comment:
Not sure if this fits here, but I wanted to drop it while you are at it: I think that "Lux" should better be what is provided by `LIGHT_LUX` ("lx" I think) to conform better with the rest of Home Assistant. 

Writing it here because "Lux" is also returned as `unit_of_measurement` by the sensor.

Reply:
Yeah, long term we want to map the device units to the Home Assistant standard units.